istresearch/floweditor
# P4-2302 - Add labels to sendmessage

# For the Reviewer


When this is complete, you should approve the PR via github.

## Summary

Added a labels field to the SendMessage action.

#### Release Note

The flow editor will now let users assign labels to outgoing messages. 

#### Breaking Changes

None.

## Quality Assurance

You have gathered the following items:


Given all of the items above, you have updated your Application ORT at the following locations:

- **Features and Alerting**: <link to app ORT>Required.
- **P4 Alerting**: <link to p4-alerting PR>Required.

## Testing and Verification

Full Stack Testing - Use RapidPro PR

React Testing - 

- Ensure you have nodeJS and NPM installed (at least version 10 and 6.4.1, respectively) 
- `sudo npm install -g yarn`
- `cd floweditor`
- `yarn install`
- In one terminal: `yarn lambda`
- In another terminal: `yarn start`

When the server is running your browser should open automatically to the URL provided by yarn start. It may take a minute or two for everything to fully come up. 

- Create a send message action
- Create several labels inside the action
- Click save - reload the page and reopen the action to ensure the settings are saved. 
- Create another send message action - the labels you already created should autocomplete.
